internal/policies/mount/mount.go

```diff
@@ -156,6 +176,135 @@ func (m *Manager) applyUserMountsPolicy(ctx context.Context, username string, en
+	}
+
+	// Marks shares to write as new units and removes from map units that shouldn't change
+	needsReload := false
+	var unitsToEnable []string
+
+	for name, content := range newUnits {
+		written, err := writeIfChanged(filepath.Join(m.unitDir, name), content)
+		if err != nil {
+			failures = append(failures, fmt.Sprintf("failed to write new unit: %v", err))
+		}
+		if written {
+			unitsToEnable = append(unitsToEnable, name)
+		}
+		needsReload = needsReload || written
+
+		delete(prevUnits, name)
+	}
+
+	// The units left in the map should be cleaned from the system.
+	unitsToClean := make([]string, 0, len(prevUnits))
+	for k := range prevUnits {
+		unitsToClean = append(unitsToClean, k)
+	}
+
+	if err = m.cleanupMountUnits(ctx, unitsToClean); err != nil {
+		failures = append(failures, fmt.Sprintf("failed when cleaning units: %v", err))
+	}
+
+	if !needsReload {
+		if failures != nil {
+			err = fmt.Errorf("%s", strings.Join(failures, "\n"))
+		}
+		return err
+	}
+
+	// Trigger a daemon reload
+	if err := m.execSystemCtlCmd(ctx, "daemon-reload"); err != nil {
+		failures = append(failures, fmt.Sprintf("failed to reload daemon: %v", err))
+		return fmt.Errorf("%s", strings.Join(failures, "\n"))
+	}
+
+	// Channel to control the error messages emitted by the routines.
+	ch := make(chan string, len(unitsToEnable))
+	// Enables and starts new units.
+	for _, name := range unitsToEnable {
+		name := name
+		go func() {
+			if err := m.execSystemCtlCmd(ctx, "enable", name); err != nil {
+				ch <- fmt.Sprintf("failed to enable unit %q: %v", name, err)
+				return
+			}
+			if err := m.execSystemCtlCmd(ctx, "start", name); err != nil {
+				ch <- fmt.Sprintf("failed to start unit %q: %v", name, err)
+				return
+			}
+			ch <- ""
+		}()
+	}
+
+	for i := 0; i < len(unitsToEnable); {
+		failure := <-ch
+		if failure != "" {
+			failures = append(failures, failure)
+		}
+		i++
+	}
+
+	if failures != nil {
+		return fmt.Errorf("%s", strings.Join(failures, "\n"))
+	}
+
+	return nil
+}
+
+// createUnits formats the adsys-.mount template with the specified paths.
+func createUnits(mountPaths []string) (units map[string]string, err error) {
+	defer decorate.OnError(&err, "failed when writing requested units")
+
+	var failures []string
+	units = make(map[string]string)
+
+	for _, mp := range mountPaths {
+		opts := []string{}
+
+		// Checks if anonymous was requested
+		p := strings.TrimPrefix(mp, "[anonymous]")
+		if p != mp {
+			opts = append(opts, "users")
+		}
+
+		_, s, found := strings.Cut(p, ":")
+		if !found {
+			failures = append(failures, fmt.Sprintf("badly formatted entry %q", mp))
+			continue
+		}
+
+		// Skips the // from the path
+		s = s[2:]
+
+		content := fmt.Sprintf(unitTemplate,
+			mp,                      // Description
+			p,                       // What
+			s,                       // Where
+			strings.Join(opts, ","), // Options
+		)
+
+		name := fmt.Sprintf("adsys-%s.mount", unit.UnitNameEscape(p))
```

Comment:
I was following the same behavior as the user mounts... gvfs does not let you mount the same shared location, even though the mount is anonymous. Should the system mounts behave different? If so, wouldn't it create confusion when managing the policies?

Reply:
We took that offline yesterday FTR. The TLDR is:
- let’s have the first entry winning
- print warnings in case of duplication with different parameters (today [krb5]/no tag, later mount options?)